### PR TITLE
Update dockers

### PR DESCRIPTION
**Pull request by nurmukhametov**
This PR updates fedora docker up to 39 and fixes ubuntu 16.04 docker. It also updates cmake versions and `-v` bash var checks.

### CODE REVIEW

File: docker/fedora/Dockerfile [hunk 1]
```diff
@@ -3,44 +3,41 @@
...
 
-FROM fedora:37
+FROM fedora:39
 LABEL maintainer="Nurmukhametov, Aleksei <aleksei.nurmukhametov@intel.com>"
 SHELL ["/bin/bash", "-c"]
 
 ARG REPO=ispc/ispc
 ARG SHA=main
 
+# If you are behind a proxy, you need to configure git to use it.
```
Comment (aneshlya) [inline docker/fedora/Dockerfile:13]: This comment and the comment on line 24 should be swapped.

File: docker/ubuntu/20.04/cpu_ispc_build/Dockerfile [hunk 2]
```diff
@@ -23,6 +23,10 @@ RUN uname -a
 RUN apt-get -y update && DEBIAN_FRONTEND=noninteractive apt-get --no-install-recommends install -y wget cmake build-essential gcc g++ git python3-dev ncurses-dev libtinfo-dev ca-certificates libtbb-dev && \
```
Comment (aneshlya) [inline docker/ubuntu/20.04/cpu_ispc_build/Dockerfile:23]: Remove `cmake` from apt-get installation.

File: docker/ubuntu/16.04/cpu_ispc_build/Dockerfile [hunk 3]
```diff
@@ -20,30 +20,54 @@ ARG EXTRA_BUILD_ARG
...
+RUN apt-get -y update && \
+    apt-get -y install zlib1g-dev zlib1g && \
+    rm -rf /var/lib/apt/lists/*
+RUN wget -q --retry-connrefused --waitretry=5 --read-timeout=20 --timeout=15 -t 5 https://www.python.org/ftp/python/3.8.18/Python-3.8.18.tgz && \
+    tar xf Python-3.8.18.tgz && cd Pyth* && \
+    ./configure && make -j"$(nproc)" && make install && \
+    rm -rf /usr/local/src/*
+
+# git >= 2.7.0 is required to use --shallow-submodules.
+RUN apt-get -y update && \
```
Comment (aneshlya) [inline docker/ubuntu/16.04/cpu_ispc_build/Dockerfile:58]: Let's not use `--shallow-submodules` in this Dockerfile. It introduces installation of extra dependencies.

File: docker/fedora/Dockerfile (hunk 1 above)
Comment (nurmukhametov) [reply docker/fedora/Dockerfile]: done

File: docker/ubuntu/20.04/cpu_ispc_build/Dockerfile (hunk 2 above)
Comment (nurmukhametov) [reply docker/ubuntu/20.04/cpu_ispc_build/Dockerfile]: removed

File: docker/ubuntu/16.04/cpu_ispc_build/Dockerfile (hunk 3 above)
Comment (nurmukhametov) [reply docker/ubuntu/16.04/cpu_ispc_build/Dockerfile]: I deleted git and adjusted calls to `alloy.py`

**Review by aneshlya — APPROVED** [review]
Looks good to me. I suggest running hadolint before merging.

File: docker/ubuntu/16.04/cpu_ispc_build/Dockerfile [hunk 4]
```diff
@@ -20,30 +20,45 @@ ARG EXTRA_BUILD_ARG
...
 
 # If you are behind a proxy, you need to configure git.
-RUN if [ -v "$http_proxy" ]; then git config --global --add http.proxy "$http_proxy"; fi
+RUN if [ -v http_proxy ]; then git config --global --add http.proxy "$http_proxy"; fi
 
 WORKDIR /usr/local/src
 
+# Zlib is required to build Python3.6.
+RUN apt-get -y update && \
+    apt-get -y install zlib1g-dev zlib1g && \
```
Comment (aneshlya) [inline docker/ubuntu/16.04/cpu_ispc_build/Dockerfile:50]: Here hadolint may require usage of `--no-install-recommends`

**Review by aneshlya — APPROVED** [review]

**Review by aneshlya — APPROVED** [review]

**Pull request merged**